internal/api/apiservice/rebuild.go

```diff
@@ -114,35 +120,20 @@ type RebuildPackageDeps struct {
 	InferStub             api.StubT[schema.InferenceRequest, schema.StrategyOneOf]
 }
 
-func RebuildPackage(ctx context.Context, req schema.RebuildPackageRequest, deps *RebuildPackageDeps) (*api.NoReturn, error) {
-	t := rebuild.Target{Ecosystem: req.Ecosystem, Package: req.Package, Version: req.Version}
-	ctx = context.WithValue(ctx, rebuild.HTTPBasicClientID, deps.HTTPClient)
-	regclient := httpinternal.NewCachedClient(deps.HTTPClient, &cache.CoalescingMemoryCache{})
-	mux := rebuild.RegistryMux{
-		CratesIO: cratesreg.HTTPRegistry{Client: regclient},
-		NPM:      npmreg.HTTPRegistry{Client: regclient},
-		PyPI:     pypireg.HTTPRegistry{Client: regclient},
-	}
-	if err := populateArtifact(ctx, &t, mux); err != nil {
-		return nil, api.AsStatus(codes.Internal, errors.Wrap(err, "selecting artifact"))
-	}
-	signer := verifier.InTotoEnvelopeSigner{EnvelopeSigner: deps.Signer}
-	a := verifier.Attestor{Store: deps.AttestationStore, Signer: signer, AllowOverwrite: deps.OverwriteAttestations}
-	if !deps.OverwriteAttestations {
-		if exists, err := a.BundleExists(ctx, t); err != nil {
-			return nil, api.AsStatus(codes.Internal, errors.Wrap(err, "checking existing bundle"))
-		} else if exists {
-			return nil, api.AsStatus(codes.AlreadyExists, errors.New("conflict with existing attestation bundle"))
-		}
-	}
-	var manualStrategy, strategy rebuild.Strategy
-	var buildDefLoc rebuild.Location
+type StrategyAndOrigin struct {
```

Comment:
Let's actually tweak this. We want to unconditionally return the Strategy but only conditionally return the repo-based one. What if we narrow this construct to just the manual and the Location. Then the getStrategy return type can be `(rebuild.Strategy, *repoStrategy, error)`. wdyt?

Reply:
I like that split, done.